docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,176 @@
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` HA cluster running version 1.9.0 or higher in order to use the process described here.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md) carefully.
+- Note that `kubeadm upgrade` will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up anything important to you. For example, any application-level state, such as a database and application might depend on (like MySQL or MongoDB) should be backed up beforehand.
+- Read [Upgrading/downgrading kubeadm clusters between v1.8 to v1.9](/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/) to learn about the relevant prerequisites.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Preparation
+
+Some preparation is needed prior to starting the upgrade. First download the version of `kubeadm` that matches the version of Kubernetes that you are upgrading to:
+
+```shell
+# Use the latest stable release or manually specify a
+# released Kubernetes version
+export VERSION=$(curl -sSL https://dl.k8s.io/release/stable.txt) 
+export ARCH=amd64 # or: arm, arm64, ppc64le, s390x
+curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /tmp/kubeadm
+chmod a+rx /tmp/kubeadm
+```
+
+Copy this file to `/tmp` on your primary master if necessary. Run this command for checking prerequisites and determining the versions you will receive:
+
+```shell
+/tmp/kubeadm upgrade plan
+```
+
+If the prerequisites are met you'll get a summary of the software versions kubeadm will upgrade to, like this:
+
+    Upgrade to the latest stable version:
+
+    COMPONENT            CURRENT   AVAILABLE
+    API Server           v1.9.0    v1.9.2
+    Controller Manager   v1.9.0    v1.9.2
+    Scheduler            v1.9.0    v1.9.2
+    Kube Proxy           v1.9.0    v1.9.2
+    Kube DNS             1.14.5    1.14.7
+    Etcd                 3.2.7     3.1.11
+
+**Caution:** Currently the only supported configuration for kubeadm HA clusters requires the use of an externally managed etcd cluster upgrading etcd is not supported as a part of the upgrade. If necessary you will have to upgrade the etcd cluster according to [etcd's upgrade instructions](/docs/tasks/administer-cluster/configure-upgrade-etcd/), which is beyond the scope of these instructions.
+{: .caution}
+
+## Upgrading your control plane
+
+The following procedure must be applied on a single master node and repeated for each subsequent master node sequentially.
+
+Before initiating the upgrade with `kubeadm` `configmap/kubeadm-config` needs to be modified for the current master host. Replace any hard reference to a master host FQDN with the current master hosts' FQDN:
+
+```shell
+kubectl get configmap -n kube-system kubeadm-config -o yaml >/tmp/kubeadm-config-cm.yaml
```

Comment:
This generally looks good to me and is the workaround I've been using for a few weeks now.

I do find referring to the `nodeName` as a FQDN potentially a bit misleading, since it does not necessarily have to be a FQDN. Maybe just refer to it as the `nodeName`?

Reply:
Good point, I've changed this (it slipped my attention because the node names happen to be FQDNs on my system).